File: 1984. Minimum Difference Between Highest and Lowest of K Scores.cpp
class Solution
{
public:
    int minimumDifference(vector<int> &nums, int k)
    {
        sort(begin(nums), end(nums));

        int ans = INT_MAX, n = nums.size();
        for (int i = 0; i <= n - k; i++)
            ans = min(ans, (nums[i + k - 1] - nums[i]));

        return ans;
    }
};

File: 1984. Minimum Difference Between Highest and Lowest of K Scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        return min([nums[i + k - 1] - nums[i] for i in range(len(nums) - k + 1)])


class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        ans, n = 10**10, len(nums)
        for i in range(n - k + 1):
            ans = min(ans, nums[i + k - 1] - nums[i])
        return ans